### PR TITLE
Remove the events that have the below criteria

### PR DESCRIPTION
**Pull request by Takishipp**
- the event's constructor is used one time.
- it use print and delete.

### CODE REVIEW

File: src/instr/instr_paje_trace.cpp [hunk 1]
```diff
@@ -331,26 +309,18 @@ void DefineEventTypeEvent::print() {
...
-DefineLinkTypeEvent::DefineLinkTypeEvent(type_t type, type_t source, type_t dest)
+void DefineLinkTypeEvent(type_t type, type_t source, type_t dest)
 {
-  this->event_type                         = PAJE_DefineLinkType;
-  this->timestamp                          = 0;
-  this->type   = type;
-  this->source = source;
-  this->dest   = dest;
+  e_event_type event_type                         = PAJE_DefineLinkType;
+  double timestamp                          = 0;
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: Please remove this variable `timestamp`, and all other timestamp variables that are not used but in XBT_DEBUG calls.

File: src/instr/instr_paje_trace.cpp [hunk 2]
```diff
@@ -218,23 +212,16 @@ void TRACE_paje_end() {
...
 }
 
-DefineContainerEvent::DefineContainerEvent(type_t type)
+void DefineContainerEvent(type_t type)
 {
-
-  event_type                            = PAJE_DefineContainerType;
-  timestamp                             = 0;
-  this->type = type;
+  e_event_type event_type                            = PAJE_DefineContainerType;
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: I think that all event_type variables can be removed: you can use its value directly now (here: PAJE_DefineContainerType)

File: src/instr/instr_paje_containers.cpp [hunk 3]
```diff
@@ -138,7 +138,7 @@ container_t PJ_container_new (const char *name, e_container_types kind, containe
   newContainer->children = xbt_dict_new_homogeneous(nullptr);
   if (newContainer->father){
     xbt_dict_set(newContainer->father->children, newContainer->name, newContainer, nullptr);
-    new CreateContainerEvent(newContainer);
+    CreateContainerEvent(newContainer);
```
Comment (mquinson) [inline src/instr/instr_paje_containers.cpp]: Please rename this function to `LogContainerCreation`

File: src/instr/instr_paje_containers.cpp [hunk 4]
```diff
@@ -206,7 +206,7 @@ void PJ_container_free (container_t container)
   if (not TRACE_disable_destroy() && container != PJ_container_get_root()) {
     //do not trace the container destruction if user requests
     //or if the container is root
-    new DestroyContainerEvent(container);
+    DestroyContainerEvent(container);
```
Comment (mquinson) [inline src/instr/instr_paje_containers.cpp]: Please rename this function to `LogContainerDestruction`

File: src/instr/instr_paje_trace.cpp [hunk 5]
```diff
@@ -244,27 +231,23 @@ void DefineContainerEvent::print() {
 	} else {
 		THROW_IMPOSSIBLE;
 	}
+  //--
 }
 
 
-DefineVariableTypeEvent::DefineVariableTypeEvent(type_t type)
+
+void DefineVariableTypeEvent(type_t type)
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: Please rename it to `LogVariableTypeDefinition`

File: src/instr/instr_paje_trace.cpp [hunk 6]
```diff
@@ -276,17 +259,26 @@ void DefineVariableTypeEvent::print() {
   }
 }
 
-DefineStateTypeEvent::DefineStateTypeEvent(type_t type)
+
+void DefineStateTypeEvent(type_t type)
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: Please rename it to `LogStateTypeDefinition`

File: src/instr/instr_paje_trace.cpp [hunk 7]
```diff
@@ -331,26 +309,18 @@ void DefineEventTypeEvent::print() {
   }
 }
 
-DefineLinkTypeEvent::DefineLinkTypeEvent(type_t type, type_t source, type_t dest)
+void DefineLinkTypeEvent(type_t type, type_t source, type_t dest)
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: Please rename it to `LogLinkTypeDefinition`

File: src/instr/instr_paje_trace.cpp [hunk 8]
```diff
@@ -360,25 +330,18 @@ void DefineLinkTypeEvent::print() {
   }
 }
 
-DefineEntityValueEvent::DefineEntityValueEvent (val_t value)
+void DefineEntityValueEvent (val_t value)
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: Please rename it to `LogEntityValue`

File: src/instr/instr_paje_trace.cpp [hunk 9]
```diff
@@ -303,20 +295,6 @@ DefineEventTypeEvent::DefineEventTypeEvent(type_t type)
...
-    stream << " " << type->id << " " << type->father->id << " " << type->name;
-    print_row();
-  } else if (instr_fmt_type == instr_fmt_TI) {
-    /* Nothing to do */
-  } else {
-    THROW_IMPOSSIBLE;
-  }
-}
-
 void DefineEventTypeEvent::print() {
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: Why did you leave this one unchanged?

Comment (Takishipp) [inline src/instr/instr_paje_trace.cpp]: Because it doesn't delete once it's called.

File: src/instr/instr_paje_trace.cpp [hunk 10]
```diff
@@ -276,17 +255,24 @@ void DefineVariableTypeEvent::print() {
...
-DefineStateTypeEvent::DefineStateTypeEvent(type_t type)
+
+void LogStateTypeDefinition(type_t type)
 {
-  this->event_type                        = PAJE_DefineStateType;
-  this->timestamp                         = 0;
-  this->type = type;
 
-  XBT_DEBUG("%s: event_type=%d", __FUNCTION__, (int)event_type);
+  XBT_DEBUG("%s: event_type=%d", __FUNCTION__, PAJE_DefineStateType);
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: Please kill it. Another debug is done a few lines after

File: src/instr/instr_paje_trace.cpp (hunk 9 above)
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: I don't see no difference with LogStateTypeDefinition(), right?

File: src/instr/instr_paje_trace.cpp [hunk 11]
```diff
@@ -360,25 +320,14 @@ void DefineLinkTypeEvent::print() {
...
-  print();
-	delete this;
-}
-
-
-void DefineEntityValueEvent::print() {
-  if (instr_fmt_type == instr_fmt_paje) {
-    XBT_DEBUG("%s: event_type=%d, timestamp=%.*f", __FUNCTION__, (int)event_type, TRACE_precision(), timestamp);
+if (instr_fmt_type == instr_fmt_paje) {
+    XBT_DEBUG("%s: event_type=%d, timestamp=%.*f", __FUNCTION__, PAJE_DefineEntityValue, TRACE_precision(),0.);
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: kill it, please

File: src/instr/instr_paje_trace.cpp [hunk 12]
```diff
@@ -390,26 +339,25 @@ void DefineEntityValueEvent::print() {
...
 
-  XBT_DEBUG("%s: event_type=%d, timestamp=%f", __FUNCTION__, (int)event_type, this->timestamp);
+void LogContainerCreation (container_t container)
+{
+  double timestamp                              = SIMIX_get_clock();
 
-  print();
-	delete this;
-}
+  XBT_DEBUG("%s: event_type=%d, timestamp=%f", __FUNCTION__, PAJE_CreateContainer,timestamp);
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: kill it please

File: src/instr/instr_paje_trace.cpp [hunk 13]
```diff
@@ -444,26 +392,23 @@ void CreateContainerEvent::print() {
...
-DestroyContainerEvent::DestroyContainerEvent (container_t container)
+void LogContainerDestruction(container_t container)
 {
-  this->event_type                              = PAJE_DestroyContainer;
-  this->timestamp                               = SIMIX_get_clock();
-  this->container = container;
+  double timestamp                               = SIMIX_get_clock();
 
-  XBT_DEBUG("%s: event_type=%d, timestamp=%f", __FUNCTION__, (int)event_type, this->timestamp);
+  XBT_DEBUG("%s: event_type=%d, timestamp=%f", __FUNCTION__, PAJE_DestroyContainer, timestamp);
```
Comment (mquinson) [inline src/instr/instr_paje_trace.cpp]: kill it, please

**Pull request merged**